### PR TITLE
Rework PwmChannel

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Rework fields used in native code.
- Simplification of parameters being passed and returned to/from native code.
- Add PwmChannel constructor accepting pin number.
- Move properties value check to native for efficiency.

## Motivation and Context
- Pin number is required on some platforms for internal configuration and other operations.
- New constructor added as private so it doesn't break compatibility with IoT Core API.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .sonarlint/nanoFramework.System.Device.Pwm.slconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (Ellerbach) [inline .sonarlint/nanoFramework.System.Device.Pwm.slconfig:1]: not sure this file should be here

File: .sonarlint/nanoframework_system.device.pwm/CSharp/SonarLint.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Ellerbach) [inline .sonarlint/nanoframework_system.device.pwm/CSharp/SonarLint.xml:1]: neither this one

File: .sonarlint/nanoframework_system.device.pwmcsharp.ruleset [hunk 3]
```diff
@@ -0,0 +1,365 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Ellerbach) [inline .sonarlint/nanoframework_system.device.pwmcsharp.ruleset:1]: neither this one :-)

**Review by Ellerbach — APPROVED** [review]
Looks good just remove the .sonarlint elements. (maybe adjust the .gitignore)

File: .sonarlint/nanoframework_system.device.pwmcsharp.ruleset (hunk 3 above)
Comment (josesimoes) [reply .sonarlint/nanoframework_system.device.pwmcsharp.ruleset:1]: Yes, those are the sonarcube rules file, allowing to run the analysis locally. If the dev doesn't have their extension installed it's OK. They are are harmless.

Comment (josesimoes) [reply .sonarlint/nanoframework_system.device.pwmcsharp.ruleset:1]: More details [here](https://stackoverflow.com/questions/37901733/sonar-lint-ruleset-files-gitignore)

**Pull request merged**